tools/nightly-e2e-tests/src/linter.py

```diff
@@ -0,0 +1,93 @@
+# -----------------------------------------------------------
+# Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019-2020 Datadog, Inc.
+# -----------------------------------------------------------
+
+import contextlib
+from dataclasses import dataclass
+from os.path import abspath
+
+
+@dataclass
+class CodeReference:
+    file_path: str
+    line_no: int
+    line_text: str
+
+
+@contextlib.contextmanager
+def linter_context(code_reference: CodeReference):
+    """
+    Context manager for convenient work with linter's code reference.
+    After context returns, the code reference is reseted to its previous value.
+    """
+    Linter.shared.push_context(code_reference=code_reference)
+    try:
+        yield
```

Comment:
this keyword is mostly used in Generators ...I am not sure I understand its usage here, can you please explain?

Reply:
The main goal of this code is to define `linter_context()` context manager. There are [two ways](https://book.pythontips.com/en/latest/context_managers.html#context-managers) of defining context managers and one of them is [by using generator](https://book.pythontips.com/en/latest/context_managers.html#implementing-a-context-manager-as-a-generator) (hence the `yield` keyword that you notice is indeed coming from generator). I used generator-way as it is shorter than class-way with implementing `_enter_` and `_exit_`.

And about context manager itself, I use it as a way to manage the linter context:
```python
with linter_context(code_reference1):
   Linter.shared.emitWarning('foo')  # 'foo' warning will be emitted in file-path and line-no from code_reference1 

   with linter_context(code_reference2): 
      Linter.shared.emitWarning('bar') # 'bar' warning will be emitted in code_reference2 

   Linter.shared.emitWarning('fizz') # 'fizz' warning will be emitted in code_reference1 
```